src/objects.c

```diff
@@ -2474,6 +2474,29 @@ CK_ATTRIBUTE *p11prov_obj_get_ec_public_raw(P11PROV_OBJ *key)
     return pub_key;
 }
 
+static int cmp_bn_attr(CK_ATTRIBUTE *x1, CK_ATTRIBUTE *x2)
```

Comment:
can you pass P11PROV_OBJ *key as argument here, so the:
        x1 = p11prov_obj_get_attr(pub_key1, CKA_MODULUS);
        x2 = p11prov_obj_get_attr(pub_key2, CKA_MODULUS);

invocation are done within this function?

I want to keep the signatures of cmp_attr and comp_bn_attr consistent

Otherwise LGTM